pkg/resource/snapshot/copy_snapshot.go

```diff
@@ -0,0 +1,185 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package snapshot
+
+import (
+	"context"
+	svcapitypes "github.com/aws-controllers-k8s/memorydb-controller/apis/v1alpha1"
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	svcsdk "github.com/aws/aws-sdk-go/service/memorydb"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (rm *resourceManager) customCreateSnapshot(
+	ctx context.Context,
+	r *resource,
+) (*resource, error) {
+	if r.ko.Spec.SourceSnapshotName != nil {
+		if r.ko.Spec.ClusterName != nil {
+			return nil, awserr.New("InvalidParameterCombinationException", "Cannot specify CacheClusteId"+
```

Comment:
Please just return a regular go error using `errors.New()`. We reserve `awserr` for errors returned by the APIs.